pvacseq/lib/generate_fasta.py

```diff
@@ -11,70 +14,134 @@ def distance_from_start(position, string):
         mutation_position = one_flanking_sequence_length
     else:
-        sys.exit("Something went wrong during the retrieval of the wildtype sequence at position(%s, %s, %s)" % chromsome, start, stop)
+        sys.exit("ERROR: Something went wrong during the retrieval of the wildtype sequence at position(%s, %s, %s)" % line['chromsome_name'], line['start'], line['stop'])
 
     return mutation_position, wildtype_subsequence
 
+def get_frameshift_subsequences(position, full_wildtype_sequence, peptide_sequence_length, line):
+    one_flanking_sequence_length = determine_flanking_sequence_length(len(full_wildtype_sequence), peptide_sequence_length, line)
+    if position < one_flanking_sequence_length:
+        start_position = 0
+    else:
+        start_position = int(position - one_flanking_sequence_length)
+    wildtype_subsequence_stop_position = int(position + one_flanking_sequence_length)
+    mutation_subsequence_stop_position = int(position)
+    wildtype_subsequence = full_wildtype_sequence[start_position:wildtype_subsequence_stop_position]
+    mutation_start_subsequence = full_wildtype_sequence[start_position:mutation_subsequence_stop_position]
+    return wildtype_subsequence, mutation_start_subsequence
 
 def main(args_input = sys.argv[1:]):
     parser = argparse.ArgumentParser("Generate Variant Sequences", description='')
     parser.add_argument('input_file', type=argparse.FileType('r'), help='input list of variants',)
     parser.add_argument('peptide_sequence_length', type=int, help='length of the peptide sequence')
     parser.add_argument('output_file', type=argparse.FileType('w'), help='output FASTA file')
-
     args = parser.parse_args(args_input)
 
-    tsvin = csv.reader(args.input_file, delimiter='\t')
-
-    pattern = re.compile('([A-Z])(\d+)([A-Z])');
+    peptide_sequence_length = args.peptide_sequence_length
+    tsvin                   = csv.DictReader(args.input_file, delimiter='\t')
+    transcript_count        = {}
+    pattern                 = re.compile('([A-Z])(\d+)([A-Z])');
     for line in tsvin:
-        match = pattern.match(line[7]);
-        if match is not None:
-            wildtype_amino_acid, position, mutant_amino_acid = match.group(1, 2, 3)
-            position = int(position) - 1
-
-            wildtype_sequence = line[9];
-            if wildtype_amino_acid != wildtype_sequence[position]:
-                continue
+        consequence_string = line['consequences']
+        if consequence_string is None:
+              continue
+        consequences = {consequence.lower() for consequence in consequence_string.split('&')}
+
+        full_wildtype_sequence = line['wildtype_amino_acid_sequence']
+        if 'frameshift_variant' in consequences:
+            consequence = 'FS'
+            position = int(line['protein_position']) - 1
+        elif 'missense_variant' in consequences or 'inframe_insertion' in consequences:
```

Comment:
Checking this here, and then immediately re-checking it on the lines below seems less than ideal. Perhaps you could store the result somehow? But I can't think of a super elegant way to handle it offhand.